crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -432,10 +385,10 @@ mod tests {
         );
 
         assert_eq!("foo", foo_module.name());
-        assert_eq!(&src, foo_module.search_path().path());
+        assert_eq!(*src, foo_module.search_path());
```

Comment:
You can add one more `PartialEq` implementation to avoid all the stars here

```rust
impl PartialEq<ModuleResolutionPathRef<'_>> for &FileSystemPath {
    fn eq(&self, other: &ModuleResolutionPathRef) -> bool {
        *self == other
    }
}
```